src/app/background.ts

```diff
@@ -21,6 +21,36 @@ const handlers = {
   getResult: (message) => tabs.get(message.tabId),
 };
 
+chrome.tabs.onUpdated.addListener((tabId, changeInfo, tab) => {
+  if (changeInfo.status === "complete" && tab?.status === "complete" && tab.url !== undefined) {
+    chrome.tabs.executeScript(
+      tabId,
+      { code: "document.documentElement.outerHTML.includes('__typename')" },
```

Comment:
```suggestion
      { code: "[...document.querySelectorAll('script')].some((script) => script.textContent.includes('__typename'))" },
```

What if we checked inside `script` tags only? I am not sure where else SSR content might be stored. I figured if a blog post or tutorial mentioned `__typename`, that might be too broad of a heuristic to just check if the whole HTML contained that string

Reply:
I think checking within script is a good call